docs/content/docs/deployment/tasks-scheduling/balanced_tasks_scheduling.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Balanced Tasks Scheduling
+weight: 5
+type: docs
+
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Balanced Tasks Scheduling
+
+This page describes the background and principle of balanced tasks scheduling, 
+how to use it when running streaming jobs.
+
+## Background
+
+When the parallelism of all vertices within a Flink streaming job is inconsistent,
```

Comment:
I wonder whether we could start by saying something about how we can get / why we would often want inconsistent parallelism in the vertices. Maybe ... 

 Flink Streaming jobs can have consistent parallelism across all vertices,  , this is useful / can occur when .... 
Different vertices in stream job can have different parallelism across their vertices, this can occur when ...... ( for example parallelism Kafka sources  may have different numbers of partition number; subsequent joining and aggregations parallelism are likely to be different again).

Reply:
> which could mean that we have 2 kafka sources with different number of partitions - I guess this change would not effect them

This algorithm only focuses on balancing the number of tasks, and specific cases require individual discussion.

> Could this be a badly configured job, if not it would be useful to detail when this can occur.

The business scenarios faced by Flink jobs are complex, so whether the configuration is reasonable depends on whether the job's performance meets the throughput requirements for processing business data.
Therefore, we cannot simply discuss whether the parallelism is properly configured.

> Are there other topologies that this effects?

No, its are not.


CC @davidradl

Reply:
That makes sense: I would expand a bit around inconsistent, mentioning parallelism. The worry is that it could be read that inconsistent parallelism requires the use of this strategy, which is not always the case.

Reply:
@davidradl 
Precise!
What about add a `note` item content like follows?  

```
<span class="label label-info">Note</span> The presence of inconsistent parallelism does not imply that this strategy must be used, as this is not always the case in practice.

```